plugins/cost-insights/src/index.ts

```diff
@@ -21,5 +21,4 @@ export * from './components';
 export { useCurrency } from './hooks';
 export * from './types';
 export * from './utils/tests';
-export * from './utils/duration';
```

Comment:
Hm, we use `intervalsOf`, `inclusiveStartDateOf` & `inclusiveEndDateOf` in several places in BIN, including our graphql client. Is there an urgent need to undo this export ?

Reply:
Internally, each of those utilities are  only used in the client (and testing the client). 

`intervalsOf` is only exported bc the plugin does not calculate intervals for `getProductInsights` as it does for other methods. This PR removes the need for exporting this utilitiy by calculating the interval internally and passing the interval via args.

`inclusiveStartDateOf` & `inclusiveEndDateOf` are exported publicly as convenience bc our graphql API implementation doesn't support tuples. Arguably, we shouldn't be exporting anything or designing the API around a particular implementation, even our own. This may require to some extent, minor code duplication.

Reply:
Even more philosophically, utility functions are implementation details that are subject to change at any time. Any change to the API of a publicly exported utility, component, etc. is a change for the whole plugin. 

The larger our footprint is, the more breaking changes we expose users to, which we should be very deliberate about and mindful of.

Reply:
@ryanmvazquez hm, yeah this is somewhat of a philosophical decision we need to be explicit about. I think it makes sense to be cautious about what we publicly export, just need to come to terms w/ the duplication for BIN :/ Side note, I think we can also remove `useCurrency` here